src/pages/Admin/OrderManagement/index.js

```diff
@@ -1,15 +1,54 @@
-import React, { useState } from 'react'
-import { Button, Col, Row, Space } from 'antd'
-import { BicycleManageAction, TableManagement } from '../../../components'
-import {
-  DeleteOutlined,
-  EditOutlined,
-  InfoCircleOutlined,
-} from '@ant-design/icons'
+import React, { useEffect, useState } from 'react'
+import { Button, Col, message, Row } from 'antd'
+import { TableManagement } from '../../../components'
+import { InfoCircleOutlined } from '@ant-design/icons'
+import OrderDetailModal from '../../../components/Admin/OrderDetailModal'
+import { useDispatch, useSelector } from 'react-redux'
+import { fetchCheckouts, updateCheckout } from '../../../redux/slices'
+import { checkoutDataSelector } from '../../../redux/selectors'
+import _ from 'lodash'
 
 function OrderManagement(props) {
   //Initialization
-  const [selectedKey, setSelectedKey] = useState([])
+  const checkoutData = useSelector(checkoutDataSelector)
+  const [showDetail, setShowDetail] = useState(false)
+  const [checkouts, setCheckouts] = useState([])
+  const [checkoutDetail, setCheckoutDetail] = useState({})
+  const [loading, setLoading] = useState(false)
+
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    setLoading(true)
+    setTimeout(() => {
+      dispatch(fetchCheckouts())
+    }, 500)
+  }, [])
+  useEffect(() => {
+    if (!_.isEmpty(checkoutData)) {
+      const checkouts = []
+      checkoutData.forEach((checkout) => {
```

Comment:
I think you should use map method for checkouts array, that make the code more clearly